### ISSUE TITLE
[XF] DisplayActionSheet broken on iOS

### ISSUE THREAD

**Issue by dansiegel** [PrismLibrary/Prism#649]
The existing implementation for DisplayActionSheet using ```Application.Current.MainPage.DisplayActionSheet( ... )``` results in an error on iOS and does not display the ActionSheet unless the page is the root page on the navigation stack.

**Comment by dansiegel** [comment]
Here is the Xamarin bug to follow it's status: https://bugzilla.xamarin.com/show_bug.cgi?id=42014

**Comment by dansiegel** [comment]
As a work around you can implement the following.

```cs
public class MyPageDialogService : PageDialogService
{
    public IApplicationProvider _applicationProvider { get; }
    public MyPageDialogService( IApplicationProvider applicationProvider ) : base( applicationProvider )
    {
        _applicationProvider = applicationProvider;
    }

    public override Task<string> DisplayActionSheetAsync( string title, string cancelButton, string destroyButton, params string[] otherButtons )
    {
        Page page = null;
        if( _applicationProvider.MainPage.Navigation.ModalStack.Count > 0 )
            page = _applicationProvider.MainPage.Navigation.ModalStack.LastOrDefault();
        else
            page = _applicationProvider.MainPage.Navigation.NavigationStack.LastOrDefault();

        if( page == null )
            page = _applicationProvider.MainPage;

        return page.DisplayActionSheet( title, cancelButton, destroyButton, otherButtons );
    }
}
```

@brianlagunas not sure if you think we should just update the PageDialogService to try to get the current page like this?

**Comment by brianlagunas** [comment]
Possibly. I am thinking this is related to the breaking change they made when they modified the ModalStack behavior.  Let's wait and see what Xamarin comes back with regarding the bug.

**Comment by brianlagunas** [comment]
Go ahead and submit a PR with this workaround.  I am not sure XF can get a fix ready in time for their 2.3.1 stable release (if they even consider this a bug), and this need to work when we release Prism 6.2.  We can always remove this check if necessary in a future release.
